example_output.md

```diff
@@ -0,0 +1,164 @@
+## Namespace
+
+### Mandatory fields for Namespace:
+
+- **name**: A single str
+
+### Optional fields for Namespace:
+
+- **description**: A single str
+- **major_version**: A single int
+- **minor_version**: A single int
+- **events**: A list of Event
+- **methods**: A list of Method
+- **typedefs**: A list of Typedef
+- **includes**: A list of Include
+- **structs**: A list of Struct
+- **enumerations**: A list of Enumeration
+- **properties**: A list of Property
+- **namespaces**: A list of Namespace
+
+
+## Event
+
+### Mandatory fields for Event:
+
+- **name**: A single str
+
+### Optional fields for Event:
+
+- **description**: A single str
+- **input**: A list of Argument
+
+
+## Argument
+
+### Mandatory fields for Argument:
+
+- **name**: A single str
+- **datatype**: A single str
+
+### Optional fields for Argument:
+
+- **description**: A single str
+- **arraysize**: A single int
+- **range**: A single str
+
+
+## Method
+
+### Mandatory fields for Method:
+
+- **name**: A single str
+
+### Optional fields for Method:
+
+- **description**: A single str
+- **error**: A list of Error
+- **input**: A list of Argument
+- **output**: A list of Argument
+
+
+## Error
```

Comment:
I like the idea of generating documentation from code, but to be really useful we would need to able to add descriptions in the Python file so it can be included in the generated documentation. I.e. already in code be able to explain e.g. how arraysize can be used.


```

@dataclass
class Error:
    datatype: str
    description: Optional[str] = str()
    arraysize: Optional[str] = None
    range: Optional[str] = None
```

Reply:
Yes and some of that should be possible using doc-strings as discussed with @miketsukerman near the end of #32 

It's possible to add additional attributes to the typing information in the dataclasses.
I haven't checked the details yet but I think it can be done.  Or it can be done in various ways by explicitly putting additional data into the classes, I'm sure.
Let's try a few things out and see if it becomes too 'ugly' or if it's OK.

Reply:
This is now partially included, with a few details to be fixed. See latest [example_output.md](https://github.com/gunnarx/vsc-tools/blob/poc_generate_docs/example_output.md) for an example.